### ISSUE TITLE
Postgres StatefulSet and Pod are created when using unmanaged database

### ISSUE THREAD

**Issue by thenets** [ansible/tower-operator#472]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
I created an AWX and a `Secret` for an external database, but a PostgreSQL pod is created anyway by a StatefulSet.

##### ENVIRONMENT
* AWX version: 19.2.2
* Operator version: 1.20.0
* Kubernetes version: v1.21.2+k3s1
* AWX install method: AWX Operator on k3s cluster

##### STEPS TO REPRODUCE

- Create a Namespace called `awx`
- Change https://raw.githubusercontent.com/ansible/awx-operator/0.12.0/deploy/awx-operator.yaml to create everything inside the `awx` Namespace
- Apply the `awx-operator.yaml`
- Apply the following yaml file:

```yaml
---
apiVersion: v1
kind: Secret
metadata:
  name: awx-thenets-postgres-configuration
  namespace: awx
stringData:
  host: "my-postgres.local."
  port: "5432"
  database: "awx"
  username: "awxuser"
  password: "awxpass"
  sslmode: "prefer"
  type: "unmanaged"
type: Opaque

---
apiVersion: awx.ansible.com/v1beta1
kind: AWX
metadata:
  name: awx-thenets
  namespace: awx
spec:
  service_type: clusterip
  ingress_type: none
  hostname: awx.site.com
```

##### EXPECTED RESULTS

- Should not exist any PostgreSQL pod.

##### ACTUAL RESULTS

- The PostgreSQL exist
- I also confirmed that AWX is using the external database. The `postgres` created by `awx-operator` is not been used.

```bash
# k -n awx get pods
NAME                            READY   STATUS    RESTARTS   AGE
awx-operator-545497f7d5-fn8sk   1/1     Running   0          25h
awx-thenets-64dbf6bdfb-tb5kn    4/4     Running   0          25h
awx-thenets-postgres-0          1/1     Running   0          24h
```

**Comment by shanemcd** [comment]
@tchellomello When you have a few, can you look at this?

**Comment by tchellomello** [comment]
```yaml
$ cat test.yml
---
apiVersion: v1
kind: Secret
metadata:
  name: awx-thenets-postgres-configuration
stringData:
  host: "my-postgres.local."
  port: "5432"
  database: "awx"
  username: "awxuser"
  password: "awxpass"
  sslmode: "prefer"
  type: "unmanaged"
type: Opaque
---
apiVersion: awx.ansible.com/v1beta1
kind: AWX
metadata:
  name: awx-thenets
spec:
  service_type: clusterip
  ingress_type: none
  hostname: awx.site.com


$ kubectl apply -f test.yml                                            14:26:43
secret/awx-thenets-postgres-configuration created
awx.awx.ansible.com/awx-thenets created

```

@thenets  I was not able to reproduce it against the `devel` 

```yaml
kubectl get awx awx-thenets -o yaml | kubectl neat                   14:27:50
apiVersion: awx.ansible.com/v1beta1
kind: AWX
metadata:
  labels:
    app.kubernetes.io/component: awx
    app.kubernetes.io/managed-by: awx-operator
    app.kubernetes.io/name: awx-thenets
    app.kubernetes.io/operator-version: devel
    app.kubernetes.io/part-of: awx-thenets
  name: awx-thenets
  namespace: default
spec:
  admin_user: admin
  create_preload_data: true
  garbage_collect_secrets: false
  hostname: awx.site.com
  image_pull_policy: IfNotPresent
  ingress_type: none
  loadbalancer_port: 80
  loadbalancer_protocol: http
  projects_persistence: false
  projects_storage_access_mode: ReadWriteMany
  projects_storage_size: 8Gi
  replicas: 1
  route_tls_termination_mechanism: Edge
  service_type: clusterip
  task_privileged: false


$ kubectl get pods -w | grep awx                           14:27:10
awx-operator-df789fd9c-rqn2k                               1/1     Running       0          62s
awx-thenets-64dbf6bdfb-qb4cg                               0/4     Pending             0          0s
awx-thenets-64dbf6bdfb-qb4cg                               0/4     Pending             0          0s
awx-thenets-64dbf6bdfb-qb4cg                               0/4     ContainerCreating   0          0s
awx-thenets-64dbf6bdfb-qb4cg                               4/4     Running             0          2s

$ kubectl iexec awx-thenets /bin/bash                                  14:32:06
Container: ✔ awx-thenets-web
bash-4.4$ cat /etc/tower/conf.d/credentials.py 
DATABASES = {
    'default': {
        'ATOMIC_REQUESTS': True,
        'ENGINE': 'awx.main.db.profiled_pg',
        'NAME': "awx",
        'USER': "awxuser",
        'PASSWORD': "awxpass",
        'HOST': 'my-postgres.local.',
        'PORT': "5432",
        'OPTIONS': { 'sslmode': 'prefer',
                     'sslrootcert': '/etc/pki/tls/certs/ca-bundle.crt',
        },
    }
}

BROADCAST_WEBSOCKET_SECRET = "E6IhwqRBrJmhsMUh69PbDUljEB6wAIKF"
```

@thenets  I'll ping you so we can troubleshoot this together

**Comment by thenets** [comment]
@tchellomello found the problem.

The issue was related to time between actions during the sequence that I followed. I was creating the environment for the first time, so:
1. First I deployed the `awx-operator`
2. Then I deployed the `awx`, without any secret (here's the problem)
3. Finally, a few minutes later, I created the `Secret` for external database, because that information was down in the main `README.md`

That means:
1. `awx-operator` created the `awx` and `postgres` containers cause doesn't existed any `Secret` for `postgres` yet
2. `awx-operator` changed the `awx` to use the external database, but don't deleted/changed the older `postgres` instance.

In short, the problem was the bigger delay between I creating the `awx` and the `Secret`.
